src/daft-local-execution/src/channel.rs

```diff
@@ -1,121 +1,126 @@
-use std::sync::Arc;
+use loole::SendError;
 
-use crate::{
-    pipeline::PipelineResultType,
-    runtime_stats::{CountingReceiver, CountingSender, RuntimeStatsContext},
-};
+pub(crate) type Sender<T> = loole::Sender<T>;
+pub(crate) type Receiver<T> = loole::Receiver<T>;
 
-pub type Sender<T> = tokio::sync::mpsc::Sender<T>;
-pub type Receiver<T> = tokio::sync::mpsc::Receiver<T>;
-
-pub fn create_channel<T>(buffer_size: usize) -> (Sender<T>, Receiver<T>) {
-    tokio::sync::mpsc::channel(buffer_size)
-}
-
-pub struct PipelineChannel {
-    sender: PipelineSender,
-    receiver: PipelineReceiver,
+pub(crate) fn create_channel<T>(buffer_size: usize) -> (Sender<T>, Receiver<T>) {
+    loole::bounded(buffer_size)
```

Comment:
@kevinzwang made a great callout the other day that the round-robin dispatching could be inefficient if ordering is not required. This can be fixed using https://docs.rs/loole/latest/loole/ channels, which are multi-producer multi-consumer. This essentially makes it work-stealing if maintaining order is not required.
```
daft.context.set_execution_config(enable_native_executor=True, default_morsel_size=1)


@daft.udf(return_dtype=daft.DataType.int64())
def do_work(x):
    x = x.to_pylist()
    # my laptop has 12 cores, so 12 workers will be spawned to run this udf.
    if x[0] % 12 == 0:
        print("doing a lot of work on ", x)
        time.sleep(1)
    else:
        print("doing a little work on ", x)
        time.sleep(0.1)
    return x


daft.from_pydict({"a": [i for i in range(48)]})
    .with_column("b", do_work(col("a")))
    .agg(col("b").sum()) # agg does not require ordering
    .collect()
```
This script is 4x faster now.